towhee/pipeline/pipeline.py

```diff
@@ -0,0 +1,187 @@
+from typing import Callable
+import uuid
+
+# pylint: disable=protected-access
+# pylint: disable=redefined-builtin
+
+
+class Pipeline:
+    """Pipeline is a tool to create data transformation chains.
+    >>> from towhee.pipeline.pipeline import Pipeline
+    >>> from towhee.pipeline.operation import ops
+    >>> def f(x):
+    ...     return x + 1
+    >>> p = ( Pipeline().input(('a', 'b'))
+    ...     .map('a', 'c', ops.towhee.decode('a', b ='b'))
+    ...     .flat_map(('a', 'b'), 'd', lambda x, y: x + y)
+    ...     .filter(('a', 'b'), ('z', 'y'), 'b', f)
+    ...     .time_window(('a', 'b'), ('w', 'x'), 'b', 1, 1, lambda x: x > 5)
+    ...     .batch('a', (('f', 'g'), 'h'), 1, 1, lambda x: x)
+    ...     .batch_all('a', 'm', f)
+    ...     .output('e')
+    ... )
+
+    >>> print(len(p._dag))
+    8
+    """
+    def __init__(self):
+        self._dag = {}
+        self._input = None
+        self._output = None
+
+    def set_config(self, config) -> 'Pipeline':
+        self._config = config
+
+    @classmethod
+    def input(cls, output_schema) -> 'Pipeline':
+        uid = uuid.uuid4().hex
```

Comment:
why not give a special name, such as `_ start`, then we can find the start node easily